File: OpenGLC.MVC/ClientApp/src/app/Models/MealEventDedtailsModel.ts
export interface MealEventDedtailsModel {
  id: string;
  eventDate: string;
  glcLevel: number;
  pospandrial: boolean;
  mealList: EventMealItemsModel[];
  mealTypeText: string;
}

export interface EventMealItemsModel {
  mealName: string;
  mealID: string;
  quantity: number;
}

File: OpenGLC.Models/Responses/MealEventDedtailsModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace OpenGLC.Models.Responses
{
	public class MealEventDedtailsModel
	{
		public MealEventDedtailsModel()
		{
			MealList = new List<EventMealItemsModel>();
		}
		public Guid Id { get; set; }
		public DateTime EventDate { get; set; }
		public int GlcLevel { get; set; }
		public bool Pospandrial { get; set; }
		public List<EventMealItemsModel> MealList { get; set; }
		public string MealTypeText { get; set; }
	}

	public class EventMealItemsModel
	{
		public string MealName { get; set; }
		public Guid MealID { get; set; }
		public int Quantity { get; set; }
	}
}
